### ISSUE TITLE
Adding phone number support

### ISSUE THREAD

**Issue by jrock2004** [tableau-mkt/jquery.addressfield#56]
Would be a cool feature enhancement to support phone number formatting. Thoughts? I could look at adding feature if you need someone to do the work.

**Comment by iamEAP** [comment]
Thanks for the interest @jrock2004!  It's tempting to add support for phone numbers, but I think it might be beyond the scope of this library.

Have you checked out https://github.com/jackocnr/intl-tel-input?  We've had good success in using both this plugin and the intl-tel-input plugin on the same form.

**Comment by jrock2004** [comment]
Yes, I was thinking that since phone number is part of an address in a lot of cases that it would be nice to have that support in there

**Comment by iamEAP** [comment]
Unfortunately, I don't have a lot of time to add that feature in myself, but I'm more than happy to review PRs!  Brain dump on how it could be integrated:

A good integration model with `intl-tel-input` is [the way we integrated validation support](https://github.com/tableau-mkt/jquery.addressfield/blob/master/src/jquery.addressfield.js#L374).  Meaning: do nothing if the intl-tel-input plugin is not installed.  Could probably be accomplished by just taking in a `phone` selector value in the initialization config object.

Here's a snippet we've used to synchronize the country field with the phone field on a couple of sites:

```javascript
// Synchronize the country selector with the phone field.
$country.change(function syncCountryAndPhoneFields() {
  $phone.intlTelInput('setCountry', $(this).val());
});
```
